include/power/power.h

```diff
@@ -65,6 +65,29 @@ extern unsigned char pm_idle_exit_notify;
  * @{
  */
 
+/**
+ * Power management notifier struct
+ *
+ * This struct contains callbacks that are called when the target enters and
+ * exits power states.
+ *
+ * @note These callbacks can be called from the ISR of the event
+ *       that caused the kernel exit from idling.
+ */
+struct pm_notifier {
+	sys_dnode_t _node;
```

Comment:
It seems that you use the list only one way (from beginning to end), so I'd go for sys_snode_t here